FBSimulatorControl.xcodeproj/project.pbxproj

```diff
@@ -3400,6 +3434,10 @@
 			isa = XCBuildConfiguration;
 			baseConfigurationReference = AA7E55C31CFEC5B6009209AE /* FBDeviceControlTests.xcconfig */;
 			buildSettings = {
+				FRAMEWORK_SEARCH_PATHS = (
```

Comment:
I certainly think the unit tests are of value. The issue I was addressing was that we could produce unit tests to validate that the Frameworks that you care about are included *without* requiring mocking of os version/xcode version.

`NSOperatingSystemVersion` provides this in `NSProcessInfo` and we already have the Xcode Version from the global configuration.

```
+ (NSArray<FBWeakFramework *> *)privateFrameworksForMacOSVersion:(NSOperatingSystemVersion)macOSVersion xcodeVersion:(NSDecimalNumber *)xcodeVersion {
  // Create and return the array, adding the relevant frameworks for the scenario you are adding
}

+ (NSArray<FBWeakFramework *> *)privateFrameworks
{
  return [self privateFrameworksForMacOSVersion:NSProcessInfo.processInfo.operatingSystemVersion xcodeVersion:FBControlCoreGlobalConfiguration.xcodeVersionNumber];
}
```

This gives you the flexibility of writing a Unit Test without requiring the mocking and a static method that uses the real values in the Framework Loader at runtime.